### PR TITLE
Fix reload with ipv6 host

### LINKED ISSUE #553: reload option works only with IPv4 host
I can not specify --reload option with IPv6 only --host.
`uvicorn --reload --host="::" example:app`

The server binds only to IPv4 host.

### LINKED ISSUE #791: Built address of server in logs is wrong in case of a ipv6 host
Prompted by the below issue, we are missing the brackets around the host in case of a ipv6 host, the below should be 
`http://[::]:8000`

```
❯ uvicorn app:app --host="::"
INFO:     Started server process [15903]
INFO:     Waiting for application startup.
INFO:     ASGI 'lifespan' protocol appears unsupported.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://:::8000 (Press CTRL+C to quit)
INFO:     ::1:35892 - "GET / HTTP/1.1" 200 OK
INFO:     ::1:35892 - "GET /favicon.ico HTTP/1.1" 200 OK
```

_Originally posted by @euri10 in https://github.com/encode/uvicorn/issues/782#issuecomment-700578964_

### PR DESCRIPTION
**Pull request by euri10**
fixes #553

correct the url printed in the logs if ipv6 is used, using brackets arout the ip as recommanded in https://www.ietf.org/rfc/rfc2732.txt

so fixes #791 as well

### CODE REVIEW

File: uvicorn/config.py [hunk 1]
```diff
@@ -112,6 +112,20 @@ def create_ssl_context(certfile, keyfile, ssl_version, cert_reqs, ca_certs, ciph
     return ctx
 
 
+def ipvXmessage(version=None):
```
Comment (cdeler) [inline uvicorn/config.py]: I have some concerns about the function.

First of all, it's a kind of a public function (at least the name doesn't start with `_`) and from my point of view it means that user can (technically) call the function. And from my point of view the naming is not pythonic, I'd change the signature to 

```suggestion
def _get_server_start_message(host_ip_version: int = 4) -> Tuple[str, str]:
```

File: uvicorn/config.py [hunk 2]
```diff
@@ -112,6 +112,20 @@ def create_ssl_context(certfile, keyfile, ssl_version, cert_reqs, ca_certs, ciph
...
+    if version == 6:
+        ip_repr = "%s://[%s]:%d"
+    elif version == 4:
```
Comment (cdeler) [inline uvicorn/config.py]: Having missed `version` argument, we'll get a `NameError` below, as `ip_repr` won't be defined.

I'd replace this `elif` by else:
```suggestion
    else:
```

(moreover I'd introduce an enum there, but it's might be an over-engineering)

Comment (euri10) [reply uvicorn/config.py]: Let me know what you think @cdeler 
I didn't find enums for the 4 or 6 version in the std lib ipaddress but I may have searched badly so I created one.
The only thing missing here is that the ipv6 link is not clickable but I guess it's a shell issue, will check that separetly

Comment (cdeler) [reply uvicorn/config.py]: It's fine, it's what I meant about the enum.

File: uvicorn/config.py [hunk 3]
```diff
@@ -112,6 +113,27 @@ def create_ssl_context(certfile, keyfile, ssl_version, cert_reqs, ca_certs, ciph
...
+    IPv6 = "IPv6"
+
+
+def _get_server_start_message(
+    host_ip_version: _IPKind = _IPKind.IPv4,
+) -> Tuple[str, str]:
+    if host_ip_version == _IPKind.IPv6:
+        ip_repr = "%s://[%s]:%d"
+    elif host_ip_version == _IPKind.IPv4:
+        ip_repr = "%s://%s:%d"
```
Comment (cdeler) [inline uvicorn/config.py]:
```suggestion
    if host_ip_version is _IPKind.IPv6:
        ip_repr = "%s://[%s]:%d"
    else:
        ip_repr = "%s://%s:%d"
```

**Review by cdeler — APPROVED** [review]
Everything is fine apart from one remark. If you think it's a reasonable suggestion you can commit it (but I'm ok without it)

Approved :fire:

**Pull request merged**